crypto/data_formats.go

```diff
@@ -15,41 +15,72 @@
 package crypto
 
 import (
-	"encoding/base64"
 	"fmt"
-	"strconv"
 
-	"github.com/benlaurie/objecthash/go/objecthash"
-	"github.com/google/trillian"
+	"github.com/google/certificate-transparency-go/tls"
+
+	pb "github.com/google/trillian"
 )
 
 // This file contains struct specific mappings and data structures.
-// TODO(gdbelvin): remove data-structure specific operations.
 
-// Constants used as map keys when building input for ObjectHash. They must not be changed
-// as this will change the output of hashRoot()
+type logVersion int
+
+const (
+	// LogRootV0 indicates version 0 for the canonical representation of the SignedLogRoot.
+	LogRootV0 logVersion = iota
```

Comment:
I need to think about it more, but I'm not sure this version thing will help much in the face of a future canonical-serialization change -- it's not communicated between server -> client, nor is it persistently stored, so if we end up with v0, v1, v2, v3, v4, .... in existence then do you just have to try all of them?

Reply:
I was considering moving these enums to be defined in `trillian.proto` and having the `Tree` struct declare what serialization to use. 

In order for that to be secure, the version format MUST be in a consistent place between all format versions.